### PR TITLE
chore: added srs validation for input files

### PR DESCRIPTION
**Pull request by CL-SHLOMIKONCHA**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✖                                                                        |
| Breaking change |✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   |✖                                                                        |
| Tests added     | ✔                                                                        |
| Chore            | ✔                                                                       |

Related issues: #XXX , #XXX ...
Closes #XXX ...

Further  information:

### CODE REVIEW

File: src/layers/models/fileValidator.ts [hunk 1]
```diff
@@ -79,6 +83,26 @@ export class FileValidator {
     });
   }
 
+  public async validateProjections(files: string[], originDirectory: string): Promise<void> {
+    const validProjection = '4326';
```
Comment (asafMasa) [inline src/layers/models/fileValidator.ts]: should be a constant

File: src/layers/models/layersManager.ts [hunk 2]
```diff
@@ -247,24 +247,27 @@ export class LayersManager {
...
       this.logger.error({
-        fileNames: data.fileNames,
-        originDirectory: data.originDirectory,
+        fileNames: fileNames,
+        originDirectory: originDirectory,
         msg: message,
       });
       throw new BadRequestError(message);
     }
+    await this.fileValidator.validateProjections(fileNames, originDirectory);
```
Comment (asafMasa) [inline src/layers/models/layersManager.ts:270]: have you checked GPKG files?

File: src/layers/models/fileValidator.ts [hunk 3]
```diff
@@ -79,6 +83,26 @@ export class FileValidator {
...
+    await Promise.all(
+      files.map(async (file) => {
+        const filePath = join(this.sourceMount, originDirectory, file);
+        const projection = await this.gdalUtilities.getProjection(filePath);
```
Comment (asafMasa) [inline src/layers/models/fileValidator.ts:91]: should we try catch and handle an exception to throw 500 with detailed message?

**Review by asafMasa — CHANGES_REQUESTED** [review]
Good work

File: src/layers/models/layersManager.ts (hunk 2 above)
Comment (CL-SHLOMIKONCHA) [reply src/layers/models/layersManager.ts:270]: yes

File: src/layers/models/fileValidator.ts (hunk 1 above)
Comment (CL-SHLOMIKONCHA) [reply src/layers/models/fileValidator.ts]: fixed

File: src/layers/models/fileValidator.ts (hunk 3 above)
Comment (CL-SHLOMIKONCHA) [reply src/layers/models/fileValidator.ts:91]: checked ,  gdal is already handle those errors

Comment (CL-SHLOMIKONCHA) [reply src/layers/models/fileValidator.ts:91]: with 500 status code

**Review by asafMasa — APPROVED** [review]

**Pull request merged**